openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1191,29 +1193,115 @@ Cache-Control: no-store
 }
 ```
 
+# Notification Endpoint {#notification_endpoint}
+
+This endpoint is used by the Wallet to notify the Credential Issuer of certain events for issued credentials, these events enable the Credential Issuer to take subsequent actions after issuance. The Credential Issuer needs to return the `notification_id` in the Credential Response or a Batch Credential Response for the Wallet to be able to use this Endpoint. Support for this endpoint is OPTIONAL.
+
+The Wallet MUST present to the Notification Endpoint a valid Access Token issued at the Token Endpoint as defined in (#token_endpoint). 
+
+Note: A Credential Issuer that requires a request to the Notification Endpoint MUST ensure the Access Token issued by the Authorization Server is valid at the Notification Endpoint.
+
+The notification from the Wallet is idempotent. The Credential Issuer MUST return success even if it receives multiple identical calls from the Wallet for the same `notification_id`s.
+
+The Wallet MAY retry if a notification request fails. The Credential Issuer may not receive certain notifications for a variety of reasons and therefore should take this into consideration.
```

Comment:
```suggestion
The Wallet MAY retry if the call to this endpoint fails. The Credential Issuer SHOULD pre-determine the amount of time within which it expects the notification. Therefore, even a well-formed notification from the Wallet could fail if received by the Credential Issuer after this time period. The Credential Issuer may not receive the notification, meaning it is unknown whether the Wallet successfully stored the credential or not - it is left to the Credential Issuer to decide how to proceed in this case.
```

Reply:
Not all notifications have a time period in which the issuer expects them to be sent such as the delete signal. For the second part of this paragraph I'd prefer we stick to the facts which is the issuer may not receive the notification and leave it to the implementer to determine what consequences might result if they were to miss a certain notification.

Reply:
Can we boil it down to this?
```suggestion
The Wallet MAY retry if a notification request fails. There are no guarantees that a Credential Issuer receives a notification within a certain time period or at all.
```